ml/src/train.py

```
@@ -190,6 +191,14 @@ def train_fold(
         fold_manifest, repeat, fold, cfg["evaluation"]["inner_k"]
     )
 
+    if withhold is not None:
+        kept_train = len(split["train"])
+        split, inner = withhold_from_training(split, inner, leaves=withhold)
```

Ruff (0.16.3)

[error] Undefined name `withhold_from_training`

(F821)